### PR TITLE
Add a onClick handler prop for the ScriptLink

### LINKED ISSUE #2784: Add a onClick handler prop for the ScriptLink
**Is your feature request related to a problem? Please describe.**
The service variant preference cookie should be set when a user clicks on the `ScriptLink`. Currently there is no way to hook into the click event.

**Describe the solution you'd like**
Add `onClick` prop to make the click event easy to hook into

**Describe alternatives you've considered**
N/A

**Testing notes**
[Tester to complete]

Dev insight: Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by ghost**
Resolves #2784

**Overall change:** _Add a click handler prop to allow us hook into the click event of the `ScriptLink`._

**Code changes:**

- _Add `onClick` prop to allow us to hook into the click event. This can be used to set the preferred service variant cookie._

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated jest tests added (for new features) or updated (for existing features)
- [ ] This PR requires manual testing

### CODE REVIEW

**Review by jroebu14 — APPROVED** [review]
Nicely done

**Review by DenisHdz — APPROVED** [review]
LGTM 👍

**Review by tochwill — APPROVED** [review]

File: packages/components/psammead-script-link/src/index.test.jsx [hunk 1]
```diff
@@ -17,18 +17,23 @@ describe('ScriptLink', () => {
...
     const links = container.querySelectorAll('a');
+    fireEvent.click(links[0]);
 
     expect(links).toHaveLength(1);
     expect(links[0]).toHaveProperty('href');
     expect(links[0].href).toEqual('https://www.bbc.co.uk/news');
     expect(links[0].textContent).toEqual('Lat');
     expect(links[0].dataset).toHaveProperty('variant');
     expect(links[0].dataset.variant).toEqual('lat');
+    expect(onClickMock).toHaveBeenCalled();
```
Comment (jroebu14) [inline packages/components/psammead-script-link/src/index.test.jsx:47]: It's worth considering splitting these tests out https://github.com/ryanmcdermott/clean-code-javascript#single-concept-per-test

Comment (ghost) [reply packages/components/psammead-script-link/src/index.test.jsx:47]: Thanks for the link. Changes applied.

**Pull request merged**